### DISCUSSION TITLE
Automate arm64 bottles on github actions by using x86 binary?

### DISCUSSION THREAD

**Discussion by will** [Homebrew/discussions#1803] [Tap maintenance and Homebrew development] [answered]
Hi, I'm working on my own tap and using the default GitHub actions. They're pretty cool, good work on those. However I don't have access to any M1 mac to use as a private runner to build arm versions. 

I am however able to build my binary statically as x86 and it does run fine on arm macs with Rosetta. I manually took apart the x86 Catalina bottle, put the static version in place of the normal dynamic linked one, re-tar'd it up, added it to the tap release, and added it with the new sha to the formula. And it all works fine for m1 macs; they download and pour the secretly x86 bottle and the program works correctly. 

I'd like to continue doing this until GitHub gets m1 runners, but it is fairly tedious. Is there some way I can modify the actions to do this for me? Or maybe some way to say a given bottle works fine for both arm and x86 macs?

**Comment by carlocab** [comment, marked as answer]
Not that I'm aware of. We encourage users who want to run things in Rosetta to install Homebrew into `/usr/local`, so I think some users might find that a formula with an ARM bottle is actually installing an x86 build meant to run under Rosetta an unwelcome surprise.

**Reply by will** [reply]
That makes sense and I can see that. But also isn't annoying for M1 users to have to maintain two separate installations of homebrew and keep in their heads which formula work on which? I honestly don’t know which is more unwelcome, since I'm not living in the arm world yet, so can't speak from experience :/

**Reply by carlocab** [reply]
We haven't gotten too many complaints about the difficulty of maintaining two parallel `brew` installations, so I imagine it's not _too_ bad.

Rosetta also isn't installed by default but gets installed when you try to run something built for Intel, so I think it could be confusing for users when they install an ARM bottle and find that running it installs Rosetta. I suppose you could add `caveats` to your formula that your ARM bottles are actually Intel builds masquerading as an ARM one.

**Reply by will** [reply]
Ok thank you!
